pkg/dockerregistry/server/pullthroughblobstore.go

```diff
@@ -118,6 +118,10 @@ func (r *pullthroughBlobStore) proxyStat(ctx context.Context, retriever importer
 }
 
 // ServeBlob attempts to serve the requested digest onto w, using a remote proxy store if necessary.
+// Important! This function is called for GET and HEAD requests. Docker client uses[1] HEAD request
```

Comment:
Yeah, HEAD or GET effectively triggers a mirror because it helps automate mirror (just HEAD every image in the repo to clone them)